x/wasm/keeper/query_plugins.go

```diff
@@ -487,7 +487,7 @@ func WasmQuerier(k wasmQueryKeeper) func(ctx sdk.Context, request *wasmvmtypes.W
 			}
 			info := k.GetContractInfo(ctx, addr)
 			if info == nil {
-				return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownAddress, request.ContractInfo.ContractAddr)
+				return nil, wasmvmtypes.NoSuchContract{Addr: request.ContractInfo.ContractAddr}
```

Comment:
Ethan's idea - make wasmvm aware of Cosmos SDK error types

Implementing the following methods on the wasmvmtypes errors will allow them to be reported properly.

https://github.com/cosmos/cosmos-sdk/blob/master/types/errors/abci.go#L103
https://github.com/cosmos/cosmos-sdk/blob/master/types/errors/abci.go#L128

Reply:
Other approach by Alex:

* Map some known sdk errors (like these ErrUnknownAddress) to wasmvmtypes errors here https://github.com/CosmWasm/wasmd/blob/9a17505cf8b637942cad78e294aa1377d7f7b2f6/x/wasm/keeper/query_plugins.go#L59
* Define concrete sdk friendly errors in wasmd, don't use  `sdkerrors.ErrUnknownAddress` use `wasmtypes.ErrNoSuchContract`, which is an sdkerrors.Error type and registered in the mapping

Reply:
For Alex's ideas, see Register here https://github.com/cosmos/cosmos-sdk/blob/master/types/errors/errors.go

We can add a custom one that handles mapping to wasmvmtypes.

Reply:
We need the actual info of the address to form the proper wasmvm type.
Thus... we cannot use sdkerrors.Error, which doesn't have a place for that info.

We could make a custom type in x/wasm that exposes `ABCICode()` and `Codespace()` and `Error()`. This type is called "NoSuchContract{Addr}" and we map this one case to a custom wasmvm error on line 59 (above comment)